api.planx.uk/send/bops.js

```diff
@@ -1,27 +1,37 @@
-import {
-  responseInterceptor,
-  fixRequestBody,
-} from "http-proxy-middleware";
-import { adminGraphQLClient } from "../hasura";
+import { responseInterceptor } from "http-proxy-middleware";
+import { adminGraphQLClient as client } from "../hasura";
 import { useProxy } from "../server";
 import { markSessionAsSubmitted } from "../saveAndReturn/utils";
 
-const client = adminGraphQLClient;
+const sendToBOPS = async (req, res, next) => {
+  // `/bops/:localAuthority` is only called via Hasura's scheduled event webhook now, so body is wrapped in a "payload" key
+  const { payload } = req.body;
+  if (!payload) {
+    next({
+      status: 400,
+      message: `Missing application payload data to send to BOPS`,
+    });
+  }
+
+  // confirm that this session has not already been successfully submitted before proceeding
+  const submittedApp = await checkBOPSAuditTable(payload?.planx_debug_data?.session_id);
+  if (submittedApp?.message === "Application created") {
+    res.status(200).send({
+      sessionId: payload?.planx_debug_data?.session_id,
+      bopsId: submittedApp?.id,
+      message: `Skipping send, already successfully submitted`,
+    });
```

Comment:
we still return success 200 if we skip so the Hasura event log doesn't show an error or failed event.